### DISCUSSION TITLE
`beet write` continuously writes albumtype on every invocation

### DISCUSSION THREAD

**Discussion by intelfx** [beetbox/beets#4648] [Q&A] [answered]
```
$ beet version
beets version 1.6.1
Python version 3.10.9
plugins: alternatives, chroma, convert, discogs, edit, embedart, fetchart, fromfilename, inline, lastimport, lyrics, missing, replaygain, scrub, unimported, zero
```

My `config.yaml`: http://ix.io/4lS3

---

I've got a pretty old beets library comprised of mostly flac and some m4a files that was last touched somewhere around 2018. After accessing it with uptodate beets, every `beet write` goes through all tracks updating the albumtype property, with the original value being empty string and the new value being seemingly correct:

```
$ beet write  
Access to Arasaka - void(); - *strtok()
  albumtype:  -> album
Access to Arasaka - void(); - kill_recorder=$c1
  albumtype:  -> album
<...>
```

On the next invocation exactly same thing happens (probably meaning that albumtype is never actually written).

What do I do with this? Is this a sign of bad configuration, or a broken database, or am I unknowingly using an unsupported combination of features, or is it a bug?

**Comment by JOJ0** [comment, marked as answer]
You've hit an ugly issue that is, if I'm not mistaken, already being worked on in https://github.com/beetbox/beets/pull/4582
HTH

**Reply by JOJ0** [reply]
This fix is merged. Please unmark this answer if the problem persists @intelfx. Thank you. HTH.

**Comment by JOJ0** [comment]
@intelfx I recently merged that fix. Is the problem gone now? Thanks!

**Comment by gamoeri** [comment]
Didn't want to start another thread so posting here - I assume I'll need to use a updated version of beets and then fix the tags (By reimporting all the affected files) ? If so, how do I update my beets? I can't find it in the docs. I downloaded beets by doing a pip install beets.

**Reply by arsaboo** [reply]
You can unistall the previous version and install directly from git using `python3 -m pip uninstall beets -y && python3 -m pip install git+https://github.com/beetbox/beets.git@master`

**Reply by JOJ0** [reply]
A full reimport might not be required. Follow these instructions: https://github.com/beetbox/beets/pull/4582#issuecomment-1445023493

Please report back if it worked out as described or could be improved. Thanks!

**Reply by gamoeri** [reply]
Uninstall/install: Worked as expected.

Reimporting MB files: It didn't rewrite some files that I wanted rewritten. 
Zeroing the other files: Worked as expected. 

There were two issues with reimporting MB files.
1) 'albumtypes::^\[' ^data_source:musicbrainz didn't find find my albumtypes field with an "e; p" in it, so I zeroed it out manually. 
2) One of my albums is not consolidated properly - a few tracks are tagged with one database, the rest with mb. When reimport that album, I merge it with the other tracks. Doing this didn't overwrite the bad tags and the tags persisted. I'm guessing the album level field doesn't let the new tag update it? I zeroed this one out too, then grouped the albums successfully. 

Thanks for the support! Greatly appreciate it.

**Reply by JOJ0** [reply]
I'd like to return that! Thanks for the detailed feedback! For 1. I could try to improve that regex to catch e; p as well

**Reply by gamoeri** [reply]
Sorry to blow this thread up but actually, reimporting all my tags and it doesn't fix the issue. I manually zeroed all the files that had the symbols in the 'albumtypes' field, then reimported all the affected files, and the semi colons appear again. I took a look at albumtype and that's fine, but the albumtypes field still populates with the symbol-y string. My most recent reimport matched with bandcamp and created the e;p albumtypes field.

**Reply by JOJ0** [reply]
U using beetcamp latest version from snejus?

**Reply by gamoeri** [reply]
I used `python3 -m pip uninstall beets -y && python3 -m pip install git+https://github.com/beetbox/beets.git@master` to reinstall

**Reply by JOJ0** [reply]
You wrote 

> My most recent reimport matched with bandcamp and created the e;p albumtypes field.

That's why I'm assuming you have _a_ bandcamp plugin installed. Which one? The IMHO currently best choice is https://github.com/snejus/beetcamp, it's a well maintained fork. Dont use the original version of the plugin, it's abandoned.

I just want to make sure we are using the latest components everywhere before moving on with analysis. To be honest, I think we should wait if things get better with the currently pending PR'S #4743 #4751 It could be totally unrelated though...not sure...

I'm currently not really working on beets things, other things in life going on. I think it would help us if you would merge in those PR branches and help testing.

**Comment by snejus** [comment]
This has been fixed in https://github.com/beetbox/beets/pull/5540 ultimately
